vxquery-core/src/main/java/org/apache/vxquery/compiler/rewriter/rules/AbstractCollectionRule.java

```diff
@@ -58,7 +60,10 @@
      *            Logical operator
      * @return collection name
      */
-    protected String[] getCollectionName(Mutable<ILogicalOperator> opRef) {
+
+    // Add functional identifies.
+    // Rename getFunctionArguments
+    protected String[] getFunctionalArguments(Mutable<ILogicalOperator> opRef, Function... functions) {
```

Comment:
Switch second argument to set<FunctionIdentifier>. Then used the contains method for doing the check.